diff --git a/features.export.inc b/features.export.inc
index cc11478..ddd39ca 100644
--- a/features.export.inc
+++ b/features.export.inc
@@ -225,7 +225,9 @@ function features_export_render($export, $module_name, $reset = FALSE) {
     // set a warning message.
     // @TODO this way of checking does not account for the possibility of inclusion instruction being commented out.
     if (isset($code['features']) && strpos($code['module'], "{$module_name}.features.inc") === FALSE) {
-      features_log(t('@module does not appear to include the @include file.', array('@module' => "{$module_name}.module", '@include' => "{$module_name}.features.inc")), 'warning');
+      // Lets add the {$module_name}.features.inc file at the end of the file.
+      // This is not as obvious as doing it as the top, but is better then assuming the user is going to add it themself.
+      $code['module'] .= "\n\ninclude_once('{$module_name}.features.inc');\n";
     }
 
     // Deprecated files. Display a message for any of these files letting the
