diff --git a/views/votingapi.views.inc b/views/votingapi.views.inc
index 63109df..859e018 100644
--- a/views/votingapi.views.inc
+++ b/views/votingapi.views.inc
@@ -286,7 +286,7 @@ function votingapi_views_data_alter(&$views_data) {
   $default_relationships[] = array(
     'description' => t('comments'),
     'entity_type' => 'comment',
-    'base_table' => 'comments',
+    'base_table' => 'comment',
     'entity_id_column' => 'cid',
     'pseudo_vote' => 'votingapi_vote',    // for legacy compatability w/RC1.
     'pseudo_cache' => 'votingapi_cache',  // for legacy compatability w/RC1.
