diff --git a/README.txt b/README.txt
index 76c8f8d..4167d0c 100644
--- a/README.txt
+++ b/README.txt
@@ -169,15 +169,14 @@ modules to be included in your install profile are noted below.
   provide:
 
   - name: The term name
-  - description: The term description
+  - description: Optional. The term description
   - weight: Optional. An explicit weight for this term
-  - vid: The vocabulary ID to which this term belongs
-  - vocab_module: Optional. Reference a vocabulary by its module rather than
-    its vid.
+  - vocabulary_machine_name: The machine name of vocabulary ID to which this
+    term belongs.
 
         terms[apples][name] = Apples
         terms[apples][description] = Delicious crunchy fruit.
-        terms[apples][vocab_module] = features_fruit
+        terms[apples][vocabulary_machine_name] = fruit
 
 
 Maintainer
diff --git a/profiler_api.inc b/profiler_api.inc
index d535d0d..c355fe7 100644
--- a/profiler_api.inc
+++ b/profiler_api.inc
@@ -336,24 +336,25 @@ function profiler_install_nodes($nodes, $config, &$identifiers) {
  * Component install callback for 'terms'.
  */
 function profiler_install_terms($terms, $config, &$identifiers) {
-  static $vocabs, $vocabs_by_module;
+  static $vocabs, $vocabs_by_machine_name;
   if (!isset($vocabs)) {
     $vocabs = taxonomy_get_vocabularies();
     foreach ($vocabs as $vid => $vocab) {
-      $vocabs_by_module[$vocab->module][$vid] = $vocab;
+      $vocabs_by_machine_name[$vocab->machine_name] = $vid;
     }
   }
   $weight = 0;
   foreach ($terms as $term) {
-    // Support the 'vocab_module' key for referring to a term's vocabulary by
-    // its module.
-    if (isset($term['vocab_module'], $vocabs_by_module[$term['vocab_module']]) && $vid = key($vocabs_by_module[$term['vocab_module']])) {
-      $term['vid'] = $vid;
+    // Support the 'vocabulary_machine_name' key for referring to a term's vocabulary
+    // by its machine name.
+    if (isset($term['vocabulary_machine_name'], $vocabs_by_machine_name[$term['vocabulary_machine_name']])) {
+      $term['vid'] = $vocabs_by_machine_name[$term['vocabulary_machine_name']];
     }
     // Sanity checks before creating.
     if (!empty($term['name']) && !empty($term['vid']) && isset($vocabs[$term['vid']])) {
       $term['weight'] = empty($term['weight']) ? $weight++ : $term['weight'];
-      taxonomy_save_term($term);
+      //Cast the term as an object, to allow it to be saved properly.
+      taxonomy_term_save((object) $term);
     }
   }
 }
@@ -397,4 +398,4 @@ function _profiler_role_id_from_name($role_name) {
   }
 
   return $rids[$role_name];
-}
\ No newline at end of file
+}
